captcha-plugin-draft.txt

```diff
@@ -107,6 +107,13 @@ We provisionally instantiate these with
 
 [TODO: I would really like to use something other than RSA, but haven't found
 satisfactory research on alternative blind signature schemes.]
+[Alex: It could be possible to use a DSA-style blind signature (see
+http://blog.cryptographyengineering.com/p/note-on-blind-signature-schemes.html).
+The protocol requires three steps, the additional step is required since the
+server initially sends a random element for each signature that needs to be
+blindly signed. While this is not ideal it may be possible to incorporate
+this into the structure below so that the plugin asks for 10000 elements of
+randomness from the server when it generates the nonces.]
```

Comment:
At first blush, all these scheme seemingly provide information theoretically secure blinding for the user, so that's okay.  

All the security proofs for the mint (exchange) are well worked out for RSA.  I have not examined it, but it appears the blind BLS signatures have a security proof from the mint's side too.  I donno about blind Schnorr signatures' security for the mint.

It sounds like the BLS scheme does not improve on RSA for computational performance, only bandwidth, right?  I suppose pairing is considered more risky than RSA but that's irrelevant here since the denomination keys expire.

Assuming one builds on the Taler wallet code base, the token withdrawal operations already happen asynchronously, so the user never experiences the bandwidth anyways.  In fact, there is no need to withdraw all the tokens initially anyways, if Tor approves using Taler reserves over longer periods.  

An extra round trip kills typical applications.  If they're not too numerous then you can transfer them while the user is entering the CAPTCHA, but it'll might add complexity and limit flexibility on the mint side.  Appears the mint cuts it's bandwidth for key material in half using Schnorr, assuming 1024 bit RSA key or 256 bit ECC keys, but like I said the user never experiences it in Taler.